### DISCUSSION TITLE
Measuring AC Coupled PV input

### DISCUSSION THREAD

**Discussion by chief-enginerd** [celsworth/lxp-bridge#234] [Q&A]
I might just be completely missing something, but is there a way to pull or calculate the power supplied by an AC coupled system connected to the GEN port on an EG4 18k?

**Comment by celsworth** [comment]
Sorry, I've actually no clue on this one. Hopefully someone else with an EG4 can help out.

If it's not in the already existing `p_pv` keys in the JSON output then it'll need adding but we'd need to know which registers to find it in.

**Reply by chief-enginerd** [reply]
No worries at all, I just wanted to see if I was just over looking something. It looks like all of the AC coupled information is still labeled the same as a generator rather than being grouped into the any of the pv Keys. Looking at the Data History table in EG4's monitor center the related points are labeled as genVolt, genFreq, genPower, and eGenAll.

**Reply by chief-enginerd** [reply]
Just realized I never got back to this. It looks like the registers associated with the gen port/AC coupled system are sent at the end of the first message with the input registers.They are:
121 - GenVolt - Generator voltage
122 - GenFreq - Generator frequency
123 - GenPower - Generator power
124 - Egen_day - Daily energy of generator
125 - Egen_all L - Low word of total generator energy
126 - Egen_all H - High word of total generator energy

Any chance these could be added for monitoring AC coupled systems?

**Reply by celsworth** [reply]
Yeah I should be able to get these added, no promises when though.. bit busy at the moment :)
